src/server/core/ml/tokenizer.py

```diff
@@ -0,0 +1,206 @@
+        text = text.replace(match.group(0), modified_number, 1)
+
+    email_matches = re.finditer(regexes["email"], text)
+    for match in email_matches:
+        modified_email = " <email>" + match.group(0).replace(".", "{dot}") + "</email> "
+        text = text.replace(match.group(0), modified_email, 1)
+
+    url_matches = re.finditer(regexes["url"], text)
+    for match in url_matches:
+        modified_url = " <url>" + match.group(0).replace(".", "{dot}").replace(":", "{colon}") + "</url> "
+        text = text.replace(match.group(0), modified_url, 1)
+
+    return text
+
+
+# Preprocessing name regex
+def name_preprocessing(text):
+    name_matches = re.finditer(regexes["name"], text)
+    for match in name_matches:
+
+        group_count = 0
+        replace_string = ""
+
+        if match.group(2):
+            replace_string += match.group(2) + "."
+            group_count += 1
+
+        if match.group(3):
+            if group_count != 0:
+                replace_string += "{space}"
+            replace_string += match.group(3)
+            group_count += 1
+
+        if match.group(10):
+            if group_count != 0:
+                replace_string += "{space}"
+            replace_string += match.group(10)
+            group_count += 1
+
+        if group_count > 1:
+            text = text.replace(match.group(0), " <name>" + replace_string + "</name> ", 1)
+
+    return text
+
+
+# Preprocessing function
+def preprocessing(text):
+    text = re.sub(r"\r?\n", " ", text)
+    text = phrases_preprocessing(text)
+    text = name_preprocessing(text)
+    text = more_preprocessing(text)
+
+    return text
+
+
+# Token processing #
+####################
+
+def process_token(token):
+    global en_US
+
+    # Process tagged tokens
+    if "<" in token and ">" in token:
+        token = re.sub(regexes["tag"], r"\9", token)
+        for special_character in special_characters:
+            token = token.replace(special_character[0], special_character[1])
+        return [token]
+    # Process untagged tokens
+    else:
+        ret_tokens = []
+        match = re.match(regexes["punctuation"], token)
+        for i in range(1, 5):
+            if match.group(i):
+                if i % 2 == 1:
+                    for c in match.group(i):
+                        ret_tokens.append(c)
+                else:
+                    ret_tokens.append(match.group(i))
+
+        # Split by hyphens
+        tmp_tokens = ret_tokens
+        ret_tokens = []
+        for ret_token in tmp_tokens:
+            if re.match(regexes["hyphen_joined"], ret_token):
+                for word in ret_token.split("-"):
+                    ret_tokens.append(word)
+            else:
+                ret_tokens.append(ret_token)
+
+        return ret_tokens
+
+
+def process_tokens(tokens):
+    final_tokens = []
+
+    global en_US
+    en_US = enchant.Dict("en_US")
```

Comment:
maybe make this a global parameter, to be able to easily change the dictionary and not have to look for this line